libr/bin/format/elf/elf.c

```diff
@@ -4391,90 +4400,95 @@ static bool section_matters(ELFOBJ *eo, int i, int type, ut32 shdr_size) {
 	return false;
 }
 
-static int _find_max_symbol_ordinal(const RVector *symbols) {
+static int _find_max_symbol_ordinal(const RVecRBinElfSymbol *symbols) {
 	int max = 0;
-
 	RBinElfSymbol *symbol;
-	r_vector_foreach (symbols, symbol) {
+	R_VEC_FOREACH (symbols, symbol) {
 		int ordinal = (int) symbol->ordinal;
 		if (ordinal > max) {
 			max = ordinal;
 		}
 	}
-
 	return max;
 }
 
 typedef struct elf_symbol_memory_t {
-	RVector *symbols;
+	RVecRBinElfSymbol *symbols_vec;
 	Elf_(Sym) *sym;
 } ElfSymbolMemory;
 
-static void _symbol_memory_free(ElfSymbolMemory *memory) {
-	r_vector_free (memory->symbols);
-	memory->symbols = NULL;
-	R_FREE (memory->sym);
+static void _symbol_memory_free(ElfSymbolMemory *em) {
+	// RVecRBinElfSymbol_fini (em->symbols_vec, NULL, NULL);
```

Comment:
its crashing, because em->symbols_vec is a reference to bo->symbols_vec. so in theory it should be allocated and freed by bobj.c and the plugins must only add/remove fields and assume its always ready to use.

Reply:
Oh weird, that's the reverse of what I had in mind. but that works also!